python/tank/descriptor/io_descriptor/github_release.py

```diff
@@ -0,0 +1,269 @@
+                version = self._find_latest_tag_by_pattern(versions, constraint_pattern)
+                can_fetch_more = next_url is not None
+        else:
+            # Otherwise, we can ask for the latest version from the github api
+            # directly.
+            versions, _ = self._get_github_releases(latest_only=True)
+            version = versions[0] if versions else None
+        if version is None or version == self.get_version():
+            # There is no latest release, or the latest release is this one, so return this descriptor.
+            return self
+        # If a release was found, generate a descriptor for that version and return it.
+        descriptor_dict = {
+            "organization": self._organization,
+            "repository": self.get_system_name(),
+            "version": version,
+            "type": "github_release",
+        }
+        desc = IODescriptorGithubRelease(descriptor_dict, self._sg_connection, self._bundle_type)
+        desc.set_cache_roots(self._bundle_cache_root, self._fallback_roots)
+        return desc
+
+    def get_latest_cached_version(self, constraint_pattern=None):
+        """
+        Returns a descriptor object that represents the latest version
+        that is locally available in the bundle cache search path.
+
+        :param constraint_pattern: If this is specified, the query will be constrained
+               by the given pattern. Version patterns are on the following forms:
+
+                - v0.1.2, v0.12.3.2, v0.1.3beta - a specific version
+                - v0.12.x - get the highest v0.12 version
+                - v1.x.x - get the highest v1 version
+
+        :returns: instance deriving from IODescriptorBase or None if not found
+        """
+        all_versions = self._get_locally_cached_versions()
+        version_numbers = all_versions.keys()
+
+        if not version_numbers:
+            return None
+
+        version_to_use = self._find_latest_tag_by_pattern(version_numbers, constraint_pattern)
+        if version_to_use is None:
+            return None
+
+        # make a descriptor dict
+        descriptor_dict = {
+            "organization": self._organization,
+            "repository": self.get_system_name(),
+            "version": version_to_use,
+            "type": "github_release",
+        }
+
+        # and return a descriptor instance
+        desc = IODescriptorGithubRelease(descriptor_dict, self._sg_connection, self._bundle_type)
+        desc.set_cache_roots(self._bundle_cache_root, self._fallback_roots)
+
+        log.debug("Latest cached version resolved to %r" % desc)
+        return desc
+
+    def has_remote_access(self):
+        """
+        Probes if the current descriptor is able to handle
+        remote requests. If this method returns, true, operations
+        such as :meth:`download_local` and :meth:`get_latest_version`
+        can be expected to succeed.
+
```

Comment:
In the case that a repository is renamed, GH API can give a 300 response. 
 https://developer.github.com/v3/#http-redirects
Right now this would result in the `has_remote_access` returning False.